sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/CheckpointFileManager.scala

```diff
@@ -303,9 +303,11 @@ class FileContextBasedCheckpointFileManager(path: Path, hadoopConf: Configuratio
     fc.mkdir(path, FsPermission.getDirDefault, true)
   }
 
-  override def create(path: Path): FSDataOutputStream = {
+  override def createTempFile(path: Path): FSDataOutputStream = {
     import CreateFlag._
-    fc.create(path, EnumSet.of(CREATE, OVERWRITE))
+    import Options._
+    fc.create(
+      path, EnumSet.of(CREATE, OVERWRITE), CreateOpts.checksumParam(ChecksumOpt.createDisabled()))
```

Comment:
@tdas Is there any specific reason why we disabled the Checksum?

Reply:
this is was a long time ago, so i dont really remember. but my guess is that the presence of absence of checkpoint can produce unexpected errors.... hence we were avoiding that confusion,

Reply:
Thanks for getting back on this. We hit a nasty bug in hadoop with files with Null checksum which wouldn't allow the NodeManager to start and wanted to disable this.